### PR TITLE
Update Starter to AWS SDK 2, KCL 2

### PR DESCRIPTION
**Pull request by jrehwaldt**
This includes various breaking changes, mainly about how to customize the used configuration (cf. `ClientConfigCustomizer`) as well as handle `Scheduler` (previously: `Worker`) creation.

Requested in OR-315

### CODE REVIEW

**Comment by jrehwaldt** [comment]
@andrejivanov As discussed earlier, integration tests are failing, because [kinesalite](https://github.com/mhart/kinesalite/issues/75)/[localstack](https://github.com/localstack/localstack/issues/893) do not currently support enhanced fan-out nor http/2 as far as I am informed. I wasn't yet able to get the tests working with the workarounds mentioned in the linked issues 🤷‍♂

File: build.gradle [hunk 1]
```diff
@@ -123,14 +124,14 @@ jacocoTestReport {
...
-        })
-    }
+//    afterEvaluate {
+//        classDirectories = files(classDirectories.files.collect {
+//            fileTree(dir: it, exclude: [
+//                "**/WorkerFactory**",
+//                "**/local/**"
+//            ])
+//        })
+//    }
```
Comment (jrehwaldt) [inline build.gradle]: Anyone can tell me what this was for?

Comment (tuhrig) [reply build.gradle]: This excludes the two folders from the Jacoco report, so that our test coverage looks better.

Comment (jrehwaldt) [reply build.gradle]: I commented it in again 🤷‍♂

**Pull request merged**